### PR TITLE
Use task user to reject during block

### LINKED ISSUE #16551: Do not expose who blocked an add-on
### Describe the problem and steps to reproduce it:

When an add-on gets blocked, we trigger a rejection for all affected files in order to disable them. 

### What happened?

The reviewer name of this rejection action is exposed in the developer hub.

### What did you expect to happen?

We should use the "task user" for the rejection.

### Anything else we should know?
(Please include a link to the page, screenshots and any relevant files.)

### PR DESCRIPTION
**Pull request by wagnerand**
Fix #16551.

### CODE REVIEW

File: src/olympia/blocklist/tests/test_admin.py [hunk 1]
```diff
@@ -1801,6 +1802,7 @@ def setUp(self):
         )
         self.change_url = reverse('admin:blocklist_block_change', args=(self.block.pk,))
         self.submission_url = reverse('admin:blocklist_blocklistsubmission_add')
+        self.task_user = user_factory(id=settings.TASK_USER_ID)
```
Comment (willdurand) [inline src/olympia/blocklist/tests/test_admin.py:1805]: Why is this needed here?

Comment (wagnerand) [reply src/olympia/blocklist/tests/test_admin.py:1805]: A few tests in that class call code that eventually passes by `disable_addon_for_block`, for example: https://github.com/mozilla/addons-server/blob/72d5bbaba243c88b13bad9e3d2bf410ec8134054/src/olympia/blocklist/tests/test_admin.py#L1895:L1895

Comment (willdurand) [reply src/olympia/blocklist/tests/test_admin.py:1805]: Interesting. I guess you don't need `self.task_user`?
```suggestion
        # We need the task user because some test cases eventually trigger
        # `disable_addon_for_block()`.
        user_factory(id=settings.TASK_USER_ID)
```

Comment (wagnerand) [reply src/olympia/blocklist/tests/test_admin.py]: Ok, thanks

**Review by willdurand — APPROVED** [review]
